test/test_torch.py

```diff
@@ -18021,6 +18020,48 @@ def is_view_of(self, base, other):
 
         return True
 
+    def transpose_if_not_contig(self, x, contiguous=False, dim0=0, dim1=1):
+        if contiguous:
+            return x
+        else:
+            return x.transpose(dim0, dim1)
+
+    @onlyOnCPUAndCUDA
+    def test_view_as_complex(self, device):
+        def fn(contiguous_input=True, dim0=0, dim1=1):
+            t = torch.randn(3, 4, 2, device=device)
+            c_t = t[:, :, 0] + 1j * t[:, :, 1]
+
+            input = self.transpose_if_not_contig(t, contiguous_input, dim0, dim1)
+
+            if (input.size()[-1] != 2 or
+                    input.stride()[-1] > 2):
+                self.assertRaisesRegex(
+                    RuntimeError, "Tensor must have a last dimension of size 2",
+                    lambda: torch.view_as_complex(input))
+                return
+
+            res = torch.view_as_complex(input)
+            self.assertEqual(res, self.transpose_if_not_contig(c_t, contiguous_input, dim0, dim1))
+            self.assertTrue(self.is_view_of(t, res))
+
+        fn()
+        fn(contiguous_input=False)
+        fn(contiguous_input=False, dim0=0, dim1=2)
+
+    @onlyOnCPUAndCUDA
+    def test_view_as_real(self, device):
+        def fn(contiguous_input=True):
+            t = torch.randn(3, 4, 2, device=device)
```

Comment:
Conceptually it's a little weird to me that this function doesn't start by creating a complex tensor. It could check complex tensors of 0, 1, and 3 dims, for example.

Reply:
Sorry: another special-case would be a complex tensor with zero elements.

Reply:
updated (with comments)!